irohad/ordering/impl/on_demand_ordering_service_impl.cpp

```diff
@@ -207,15 +208,12 @@ bool OnDemandOrderingServiceImpl::batchAlreadyProcessed(
   // if any transaction is commited or rejected, batch was already processed
   // Note: any_of returns false for empty sequence
   return std::any_of(
-      tx_statuses.begin(), tx_statuses.end(), [this](const auto &batch_result) {
-        return iroha::visit_in_place(
-            batch_result,
-            [](const ametsuchi::tx_cache_status_responses::Missing &) {
-              return false;
-            },
-            [this](const auto &status) {
-              log_->warn("Duplicate transaction: {}", status.hash.hex());
-              return true;
-            });
+      tx_statuses.begin(), tx_statuses.end(), [this](const auto &tx_status) {
+        if (iroha::ametsuchi::isAlreadyProcessed(tx_status)) {
```

Comment:
Don't like the resulting namespaces. Hard to understand, where and how the function `isAlreadyProcessed` will look for tx. Maybe, it's better to create a namespace of third level, so that `iroha::ametsuchi::cache_utils::isAlreadyProcessed` or make methods static and put them into distinct class?

Reply:
Since it processes tx status and not a transaction, I think its quite obvious that it does not look for tx